### PR TITLE
[#12048] Migrate GetFeedbackSessionsAction

### PR DESCRIPTION
**Pull request by daongochieu2810**
Part of https://github.com/TEAMMATES/teammates/issues/12048

Outline of Solution

Modified GetFeedbackSessionsAction to work with dual DB.

### CODE REVIEW

**Review by kevin9foong — APPROVED** [review]

File: src/main/java/teammates/storage/sqlentity/FeedbackSession.java [hunk 1]
```diff
@@ -112,6 +112,35 @@ public FeedbackSession(String name, Course course, String creatorEmail, String i
...
+     */
+    public FeedbackSession getCopyForStudent(String studentEmail) {
+        FeedbackSession copy = getCopy();
+        for (DeadlineExtension de : copy.getDeadlineExtensions()) {
+            if (!de.getUser().getEmail().equals(studentEmail)) {
+                de.setEndTime(copy.getEndTime());
+            }
+        }
+        return copy;
+    }
```
Comment (samuelfangjw) [inline src/main/java/teammates/storage/sqlentity/FeedbackSession.java:129]: We are not using this getCopyForStudent and getCopyForInstructor anymore, doubt this is necessary.

File: src/main/java/teammates/ui/webapi/GetFeedbackSessionsAction.java [hunk 2]
```diff
@@ -47,8 +51,13 @@ void checkSpecificAccessControl() throws UnauthorizedAccessException {
             }
 
             if (courseId != null) {
-                CourseAttributes courseAttributes = logic.getCourse(courseId);
-                gateKeeper.verifyAccessible(logic.getStudentForGoogleId(courseId, userInfo.getId()), courseAttributes);
+                if (isCourseMigrated(courseId) && isAccountMigrated(userInfo.getId())) {
```
Comment (samuelfangjw) [inline src/main/java/teammates/ui/webapi/GetFeedbackSessionsAction.java:54]: Shouldn't be a need to check both. Can assume accounts are migrated before courses.

File: src/main/java/teammates/ui/webapi/GetFeedbackSessionsAction.java [hunk 3]
```diff
@@ -77,13 +169,8 @@ public JsonResult execute() {
                 feedbackSessionAttributes = new ArrayList<>();
                 for (StudentAttributes student : students) {
                     String studentCourseId = student.getCourse();
-                    String emailAddress = student.getEmail();
                     List<FeedbackSessionAttributes> sessions = logic.getFeedbackSessionsForCourse(studentCourseId);
 
-                    sessions = sessions.stream()
-                        .map(session -> session.getCopyForStudent(emailAddress))
-                        .collect(Collectors.toList());
-
```
Comment (samuelfangjw) [inline src/main/java/teammates/ui/webapi/GetFeedbackSessionsAction.java:86]: Original behaviour shouldn't be changed.

File: src/test/java/teammates/ui/webapi/GetFeedbackSessionsActionTest.java [hunk 4]
```diff
@@ -190,7 +190,7 @@ protected void testExecute_asStudentWithCourseId_shouldReturnAllSessionsForCours
 
     }
 
-    @Test
+    @Test(enabled = false)
```
Comment (samuelfangjw) [inline src/test/java/teammates/ui/webapi/GetFeedbackSessionsActionTest.java:193]: Any particular reason we are ignoring these tests? Doesn't seem like a test that should fail.

**Review by samuelfangjw — CHANGES_REQUESTED** [review]

File: src/main/java/teammates/storage/sqlentity/FeedbackSession.java (hunk 1 above)
Comment (daongochieu2810) [reply src/main/java/teammates/storage/sqlentity/FeedbackSession.java:129]: I think we are still using `getCopyForStudent` in GetFeedbackSessionsAction line 100

**Comment by daongochieu2810** [comment]
@samuelfangjw the logic might seem a bit complicated here. Since the courseId could be null, when we fetch the list of course ids from the account, we need to check whether each course has been migrated or not, then carry out both old and new logic

**Comment by kevin9foong** [comment]
Hi @daongochieu2810, to merge this PR, would it be possible to update this branch with the latest v9-migration? Thank you!

**Review by samuelfangjw — APPROVED** [review]

**Pull request merged**